File: backend/internal/domain/partition_stat.go
package domain

import "fmt"

type PartitionStat struct {
	Device     string   `json:"device"`
	Mountpoint string   `json:"mountpoint"`
	Fstype     string   `json:"fstype"`
	Opts       []string `json:"opts"`
}

func NewPartitionStat(device, mountpoint, fstype string, opts []string) *PartitionStat {
	return &PartitionStat{
		Device:     device,
		Mountpoint: mountpoint,
		Fstype:     fstype,
		Opts:       opts,
	}
}

func (d PartitionStat) ToString() string {
	return fmt.Sprintf(
		"Device: %s\nMountpoint: %s\nFstype: %s\nOpts: %v\n",
		d.Device, d.Mountpoint, d.Fstype, d.Opts)
}

File: frontend/lib/models/partition_stat.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'partition_stat.freezed.dart';
part 'partition_stat.g.dart';

@freezed
class PartitionStat with _$PartitionStat {
  const factory PartitionStat({
    required String device,
    required String mountpoint,
    required String fstype,
    required List<String> opts,
  }) = _PartitionStat;

  factory PartitionStat.fromJson(Map<String, dynamic> json) =>
      _$PartitionStatFromJson(json);
}
